src/Symfony/Component/HttpKernel/DependencyInjection/AddCacheWarmerPass.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Compiler\PriorityTaggedServiceTrait;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+
+/**
+ * Registers the cache warmers.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class AddCacheWarmerPass implements CompilerPassInterface
+{
+    use PriorityTaggedServiceTrait;
+
+    private $tag;
+    private $serviceName;
+
+    public function __construct($serviceName = 'cache_warmer', $tag = 'kernel.cache_warmer')
+    {
+        if (2 > func_num_args()) {
```

Comment:
With _new class_ i meant `Symfony\Component\HttpKernel\DependencyInjection\AddCacheClearerPass` (a new class definition). It can have required args, that can be provided by the legacy class (the _old_ class) by adding a constructor...

```php
// Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler\AddCacheClearerPass
public function __construct() {
   parent::__construct('cache_warmer', 'kernel.cache_warmer');
}
```